File: WEEK3/code/boilerplate.R
#!/usr/bin/env Rscript

# Author: Y_Sun ys219@ic.ac.uk
# Script: boilerplate.R
# Desc: function exercises in r language
# Arguments: 0
# Input:Rscript boilerplate.R
# Output: printed output in r terminal
# Date: Oct 2019

# A boilerplate R script

MyFunction <- function(Arg1, Arg2){

  # Statements involving Arg1, Arg2:
  print(paste("Argument", as.character(Arg1), "is a", class(Arg1))) # print Arg1's type
  print(paste("Argument", as.character(Arg2), "is a", class(Arg2))) # print Arg2's type

  return (c(Arg1, Arg2)) #this is optional, but very useful
}

MyFunction(1,2) #test the function
MyFunction("Riki","Tiki") #A different test

File: WEEK2/code/boilerplate.py
#!/usr/bin/env python3

# Author: Y_Sun ys219@ic.ac.uk
# Script: boilerplate.py
# Desc: a boilerplate
# Arguments: python script sample
# Input:ipython3 boilerplate.py
# Output: printed statement
# Date: Oct 2019

"""Description of this program or application.
you can use several line"""

_appname_="boilerplate"
_author_="ys"
_version_="0.0.1"
_license_="code.program"

import sys

def main(argv):
    """main entry point of the program"""
    print('This is a boillerplate')
    return 0
##would able take input with argv
if __name__ == "__main__":
    """make sure the "main" function is called from command line"""
    status = main(sys.argv)
    sys.exit(status)
